pkg/apis/messaging/v1alpha1/sequence_validation_test.go

```diff
@@ -144,27 +147,28 @@ func TestSequenceSpecValidation(t *testing.T) {
 		ts: &SequenceSpec{
 			ChannelTemplate: validChannelTemplate,
 			Steps:           []SubscriberSpec{{URI: &subscriberURI}},
-			Reply: &corev1.ObjectReference{
-				APIVersion: "messaging.knative.dev/v1alpha1",
-				Kind:       "inmemorychannel",
+			Reply: &v1alpha1.Destination{
+				DeprecatedAPIVersion: "messaging.knative.dev/v1alpha1",
+				DeprecatedKind:       "inmemorychannel",
 			},
 		},
 		want: func() *apis.FieldError {
 			fe := apis.ErrMissingField("reply.name")
 			return fe
 		}(),
-	}, {
-		name: "valid sequence with invalid reply",
-		ts: &SequenceSpec{
-			ChannelTemplate: validChannelTemplate,
-			Steps:           []SubscriberSpec{{URI: &subscriberURI}},
-			Reply:           makeInvalidReply("reply-channel"),
-		},
-		want: func() *apis.FieldError {
-			fe := apis.ErrDisallowedFields("reply.Namespace")
-			fe.Details = "only name, apiVersion and kind are supported fields"
-			return fe
-		}(),
+		// TODO current destination ref allows setting the namespace, thus this fails.
```

Comment:
/hold

we have a DeprecatedNamespace in Destination, is that correct? I thought we weren't supporting namespaces in ObjectRef..
If that is the case, the Ref in Destination should actually validate that... as done [here](https://github.com/knative/eventing/blob/710f048a38484447d18d5cd30968c109fc8a7b4e/pkg/apis/messaging/v1alpha1/subscribable_channelable_validation.go#L37)

Reply:
/hold cancel, it seems that we will let this through